src/lib/storage.js

```diff
@@ -360,28 +360,33 @@ class Storage {
       uplink with proxy_access rights against {name} and combines results
       into one json object
     Used storages: local && uplink (proxy_access)
-   * @param {*} name
-   * @param {*} options
-   * @param {*} callback
-   */
-  get_package(name, options, callback) {
-    if (_.isFunction(options)) {
-      callback = options, options = {};
-    }
 
-    this.localStorage.getPackageMetadata(name, (err, data) => {
+   * @param {object} options
+   * @property {string} options.name Package Name
+   * @property {object}  options.req Express `req` object
+   * @property {boolean} options.keepUpLinkData keep up link info in package meta, last update, etc.
```

Comment:
You will need to elaborate what means `keepUpLinkData`. It seems a big change of logic here

Reply:
Line 367 already explained it, isn't it?